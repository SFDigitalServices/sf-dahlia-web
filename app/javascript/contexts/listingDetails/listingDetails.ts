interface ListingDetails {
  /**  */
  uid: string

  /**  */
  email: string

  /**  */
  created_at: Date

  /**  */
  updated_at: Date
}

interface ListingDetailsData {
  data: any
  success: boolean
}

export { ListingDetails, ListingDetailsData }
