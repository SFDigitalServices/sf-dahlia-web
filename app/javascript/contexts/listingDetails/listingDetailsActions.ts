import { createAction } from "typesafe-actions"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart } from "../../api/types/rails/listings/RailsAmiChart"

export enum ListingDetailsActions {
  StartFetchingUnits = "SetFetchingUnits",
  FinishFetchingUnits = "FinishFetchingUnits",
  SetUnits = "SetUnits",
  SetAmiCharts = "SetAmiCharts",
  StartFetchingAmiCharts = "SetFetchingAmiCharts",
  SetFetchedAmiCharts = "SetFetchedAmiCharts",
  SetFetchingAmiChartsError = "SetFetchingAmiChartsError",
  SetFetchingUnitsError = "SetFetchingUnitsError",
  FinishFetchingAmiCharts = "FinishFetchingAmiCharts",
}

export const startFetchingUnits = createAction(ListingDetailsActions.StartFetchingUnits)()
export const setUnits = createAction(ListingDetailsActions.SetUnits)()
export const setAmiCharts = createAction(ListingDetailsActions.SetAmiCharts)()
export const finishFetchingAmiCharts = createAction(ListingDetailsActions.FinishFetchingAmiCharts)<
  RailsAmiChart[]
>()
export const setFetchedAmiCharts = createAction(ListingDetailsActions.SetFetchedAmiCharts)()
export const startFetchingAmiCharts = createAction(ListingDetailsActions.StartFetchingAmiCharts)()
export const finishFetchingUnits = createAction(ListingDetailsActions.FinishFetchingUnits)<
  RailsUnit[]
>()
export const setFetchingUnitsError = createAction(
  ListingDetailsActions.SetFetchingUnitsError
)<Error>()
export const setFetchingAmiChartsError = createAction(
  ListingDetailsActions.SetFetchingAmiChartsError
)<Error>()
