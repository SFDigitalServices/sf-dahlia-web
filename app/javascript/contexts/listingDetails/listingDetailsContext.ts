import { createContext } from "react"
import RailsUnit from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart, RailsAmiChartMetaData } from "../../api/types/rails/listings/RailsAmiChart"

export type ContextProps = {
  units: RailsUnit[]
  amiCharts: RailsAmiChart[]
  fetchedAmiCharts: boolean
  fetchingAmiCharts: boolean
  fetchedUnits: boolean
  fetchingUnits: boolean
  fetchUnits: (listingId: string) => void
  fetchAmiCharts: (chartsToFetch: RailsAmiChartMetaData[]) => void
  fetchingAmiChartsError: Error
  fetchingUnitsError: Error
}

export default createContext<Partial<ContextProps>>({})
