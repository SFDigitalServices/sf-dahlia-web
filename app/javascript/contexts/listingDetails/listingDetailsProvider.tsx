import React, { useReducer } from "react"
import ListingDetailsReducer from "./listingDetailsReducer"
import ListingDetailsContext, { ContextProps } from "./listingDetailsContext"
import { getUnits, getAmiCharts } from "../../api/listingApiService"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart, RailsAmiChartMetaData } from "../../api/types/rails/listings/RailsAmiChart"
import {
  finishFetchingUnits,
  finishFetchingAmiCharts,
  startFetchingAmiCharts,
  startFetchingUnits,
  setFetchingAmiChartsError,
  setFetchingUnitsError,
} from "./listingDetailsActions"

interface ListingDetailsProviderProps {
  children?: React.ReactNode
}

const ListingDetailsProvider = (props: ListingDetailsProviderProps) => {
  const [state, dispatch] = useReducer(ListingDetailsReducer, {
    units: [],
    amiCharts: [],
    fetchingUnits: false,
    fetchedUnits: false,
    fetchingAmiCharts: false,
    fetchedAmiCharts: false,
    fetchingAmiChartsError: null,
    fetchingUnitsError: null,
  })

  const contextValues: ContextProps = {
    units: state.units,
    amiCharts: state.amiCharts,
    fetchedAmiCharts: state.fetchedAmiCharts,
    fetchingAmiCharts: state.fetchingAmiCharts,
    fetchedUnits: state.fetchedUnits,
    fetchingUnits: state.fetchingUnits,
    fetchingAmiChartsError: state.fetchingAmiChartsError,
    fetchingUnitsError: state.fetchingUnitsError,
    fetchUnits: (listingId: string) => {
      dispatch(startFetchingUnits())
      getUnits(listingId)
        .then((units: RailsUnit[]) => {
          dispatch(finishFetchingUnits(units))
        })
        .catch((error) => {
          console.error(error)
          dispatch(setFetchingUnitsError(error))
        })
    },
    fetchAmiCharts: (chartsToFetch: RailsAmiChartMetaData[]) => {
      dispatch(startFetchingAmiCharts())
      getAmiCharts(chartsToFetch)
        .then((amiCharts: RailsAmiChart[]) => {
          dispatch(finishFetchingAmiCharts(amiCharts))
        })
        .catch((error) => {
          console.error(error)
          dispatch(setFetchingAmiChartsError(error))
        })
    },
  }

  return (
    <ListingDetailsContext.Provider value={contextValues}>
      {props.children}
    </ListingDetailsContext.Provider>
  )
}

export default ListingDetailsProvider
