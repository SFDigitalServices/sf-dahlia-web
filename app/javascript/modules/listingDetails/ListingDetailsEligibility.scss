.info-card{
    --title-color:var(--bloom-color-gray-950);
}

.detail-header {
    min-height: var(--bloom-s20);
    .detail-header__hgroup {
        display: flex;
        flex-direction: column;
        justify-content: center;
    }
}

.additional-rule-card {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
}
