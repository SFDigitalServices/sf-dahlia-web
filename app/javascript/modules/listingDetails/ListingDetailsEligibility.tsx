import React from "react"
import {
  ExpandableText,
  InfoCard,
  ListingDetailItem,
  ListSection,
  StandardTable,
  t,
} from "@bloom-housing/ui-components"
import { RailsListing } from "../listings/SharedHelpers"
import { isHabitatListing, isSale } from "../../util/listingUtil"
import { renderInlineWithInnerHTML } from "../../util/languageUtil"
import { BeforeApplyingForSale, BeforeApplyingType } from "../../components/BeforeApplyingForSale"
import { ListingDetailsPreferences } from "./ListingDetailsPreferences"

export interface ListingDetailsEligibilityProps {
  listing: RailsListing
  imageSrc: string
}

export const ListingDetailsEligibility = ({
  listing,
  imageSrc,
}: ListingDetailsEligibilityProps) => {
  /* TODO: Implement updated API to get actual data */
  const HMITableHeaders = {
    householdSize: "t.householdSize",
    maxIncomeMonth: "t.maximumIncomeMonth",
    maxIncomeYear: "t.maximumIncomeYear",
  }
  const HMITableData = [
    {
      householdSize: {
        content: <span className="font-semibold">{`1 ${t("listings.person")}`}</span>,
      },
      maxIncomeMonth: { content: t("t.perMonthCost", { cost: "$1,111" }) },
      maxIncomeYear: { content: t("t.perYearCost", { cost: "$51,111" }) },
    },
    {
      householdSize: {
        content: <span className="font-semibold">{`2 ${t("listings.people")}`}</span>,
      },
      maxIncomeMonth: { content: t("t.perMonthCost", { cost: "$1,111" }) },
      maxIncomeYear: { content: t("t.perYearCost", { cost: "$51,111" }) },
    },
    {
      householdSize: {
        content: <span className="font-semibold">{`3 ${t("listings.people")}`}</span>,
      },
      maxIncomeMonth: { content: t("t.perMonthCost", { cost: "$1,111" }) },
      maxIncomeYear: { content: t("t.perYearCost", { cost: "$51,111" }) },
    },
    {
      householdSize: {
        content: <span className="font-semibold">{`4 ${t("listings.people")}`}</span>,
      },
      maxIncomeMonth: { content: t("t.perMonthCost", { cost: "$1,111" }) },
      maxIncomeYear: { content: t("t.perYearCost", { cost: "$51,111" }) },
    },
  ]

  const occupancyTableHeaders = {
    unitType: "t.unitType",
    occupancy: "t.occupancy",
  }
  const occupancyTableData = listing.unitSummaries.general.map((unit) => ({
    unitType: {
      content: <span className="font-semibold">{t(`listings.unitTypes.${unit.unitType}`)}</span>,
    },
    occupancy: { content: `${unit.minOccupancy}-${unit.maxOccupancy} ${t("listings.people")}` },
  }))
  return (
    <ListingDetailItem
      imageAlt={""}
      imageSrc={imageSrc}
      title={t("listings.eligibility.header")}
      subtitle={isSale(listing) ? "" : t("listings.eligibility.subheader")}
      desktopClass="bg-primary-lighter"
    >
      {isSale(listing) && (
        <BeforeApplyingForSale
          beforeApplyingType={
            isHabitatListing(listing)
              ? BeforeApplyingType.LISTING_DETAILS_HABITAT
              : BeforeApplyingType.LISTING_DETAILS
          }
        />
      )}
      <ListSection
        title={t("listings.householdMaximumIncome")}
        subtitle={
          <div>
            <p className="mb-4">{renderInlineWithInnerHTML(t("listings.forIncomeCalculations"))}</p>
            <p className="mb-4">
              {renderInlineWithInnerHTML(t("listings.incomeExceptions.intro"))}
            </p>
            <ul className="list-disc ml-5">
              <li>{t("listings.incomeExceptions.students")}</li>
              <li>{t("listings.incomeExceptions.nontaxable")}</li>
            </ul>
          </div>
        }
      >
        <StandardTable headers={HMITableHeaders} data={HMITableData} />
      </ListSection>
      <ListSection title={t("t.occupancy")} subtitle={t("listings.occupancyDescriptionNoSro")}>
        <StandardTable headers={occupancyTableHeaders} data={occupancyTableData} />
      </ListSection>
      <ListingDetailsPreferences listingID={listing.listingID} />
      <ListSection
        title={t("listingsForRent.rentalAssistance.title")}
        subtitle={t("listingsForRent.rentalAssitance.subtitle")}
      />
      <ListSection
        title={t("listings.additionalEligibilityRules.title")}
        subtitle={t("listings.additionalEligibilityRules.subtitle")}
      >
        {listing.Credit_Rating && (
          <InfoCard title={t("listings.additionalEligibilityRules.creditHistory")}>
            <ExpandableText className="text-sm text-gray-700">
              {listing.Credit_Rating}
            </ExpandableText>
          </InfoCard>
        )}

        {listing.Eviction_History && (
          <InfoCard title={t("listings.additionalEligibilityRules.rentalHistory")}>
            <ExpandableText className="text-sm text-gray-700">
              {listing.Eviction_History}
            </ExpandableText>
          </InfoCard>
        )}

        <InfoCard title={t("listings.additionalEligibilityRules.criminalBackground")}>
          <ExpandableText className="text-sm text-gray-700">
            Qualified applicants with criminal history will be considered for housing in compliance
            with Article 49 of the San Francisco Police Code: Fair Chance Ordinance.
          </ExpandableText>
        </InfoCard>
        <p>
          <a href={listing.Building_Selection_Criteria} target={"_blank"}>
            {t("listings.additionalEligibilityRules.findOutMore")}
          </a>
        </p>
      </ListSection>
    </ListingDetailItem>
  )
}
