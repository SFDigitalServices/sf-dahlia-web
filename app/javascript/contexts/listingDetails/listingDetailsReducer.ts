import { createReducer } from "typesafe-actions"
import { ListingDetailsActions } from "./listingDetailsActions"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart } from "../../api/types/rails/listings/RailsAmiChart"
import { AmiChartData } from "./listingDetailsContext"

type ListingDetailsState = {
  fetchingUnits: boolean
  fetchedUnits: boolean
  fetchingAmiCharts: boolean
  fetchedAmiCharts: boolean
  units: RailsUnit[]
  amiChartData: AmiChartData
  amiCharts: RailsAmiChart[]
  fetchingAmiChartsError: Error
  fetchingUnitsError: Error
}

const ListingDetailsReducer = createReducer({ units: [] } as ListingDetailsState, {
  [ListingDetailsActions.StartFetchingUnits]: (state) => {
    return { ...state, fetchingUnits: true }
  },
  [ListingDetailsActions.StartFetchingAmiCharts]: (state) => {
    return { ...state, fetchingAmiCharts: true }
  },
  [ListingDetailsActions.FinishFetchingUnits]: (state, { payload: units }) => {
    const amiChartYears: Array<number> = []
    const amiChartTypes = []
    const amiChartPercentages = []

    units.forEach((unit: RailsUnit) => {
      const amiChartYear = amiChartYears.find((year) => {
        return year === unit.AMI_chart_year
      })

      const amiChartType = amiChartTypes.find((type) => {
        return type === unit.AMI_chart_type
      })

      const amiChartPercentage = amiChartPercentages.find((type) => {
        return type === unit.Max_AMI_for_Qualifying_Unit
      })

      if (!amiChartYear) {
        amiChartYears.push(unit.AMI_chart_year)
      }

      if (!amiChartType) {
        amiChartTypes.push(unit.AMI_chart_type)
      }

      if (!amiChartPercentage) {
        amiChartPercentages.push(unit.Max_AMI_for_Qualifying_Unit)
      }
    })

    return {
      ...state,
      fetchingUnits: false,
      fetchedUnits: true,
      units,
      amiChartData: {
        years: amiChartYears,
        percentages: amiChartPercentages,
        types: amiChartTypes,
      },
    }
  },
  [ListingDetailsActions.FinishFetchingAmiCharts]: (state, { payload: amiCharts }) => {
    return {
      ...state,
      fetchingAmiCharts: false,
      fetchedAmiCharts: true,
      amiCharts: amiCharts?.map((amiChart: RailsAmiChart) => {
        return {
          ...amiChart,
          chartType: amiChart?.values[0]?.chartType,
        }
      }),
    }
  },
})

export default ListingDetailsReducer
