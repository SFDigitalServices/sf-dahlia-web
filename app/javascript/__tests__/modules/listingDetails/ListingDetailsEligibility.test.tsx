import React from "react"
import { render, cleanup } from "@testing-library/react"
import { ListingDetailsEligibility } from "../../../modules/listingDetails/ListingDetailsEligibility"
import { preferences as defaultPreferences } from "../../data/RailsListingPreferences/lottery-preferences-default"
import { closedRentalListing } from "../../data/RailsRentalListing/listing-rental-closed"
import { openSaleListing } from "../../data/RailsSaleListing/listing-sale-open"
import ListingDetailsContext from "../../../contexts/listingDetails/listingDetailsContext"
import { unitsWithOneAmi } from "../../data/RailsListingUnits/listing-units"
import { amiChartsWithOneAmi } from "../../data/RailsAmiCharts/ami-charts"
import {
  sroMixedRentalListing,
  sroRentalListing,
} from "../../data/RailsRentalListing/listing-rental-sro"
import { habitatListing } from "../../data/RailsSaleListing/listing-sale-habitat"
import { rentalEducatorListing } from "../../data/RailsRentalListing/listing-rental-educator"
import { t } from "@bloom-housing/ui-components"
import { renderAndLoadAsync } from "../../__util__/renderUtils"

const axios = require("axios")

jest.mock("axios")

describe("ListingDetailsEligibility", () => {
  beforeEach(() => {
    // This component pulls in react-media, which needs this custom mock
    window.matchMedia = jest.fn().mockImplementation((query) => {
      return {
        matches: true,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      }
    })
  })

  afterEach(() => {
    cleanup()
    jest.clearAllMocks()
    jest.resetAllMocks()
  })

  it("displays listing details eligibility section and no Building Selection Criteria Link", async (done) => {
    const testListing = {
      ...closedRentalListing,
      Building_Selection_Criteria: "",
    }

    axios.get.mockResolvedValue({ data: { preferences: defaultPreferences } })

    const { asFragment, findByText } = render(
      <ListingDetailsContext.Provider
        value={{
          units: unitsWithOneAmi,
          amiCharts: amiChartsWithOneAmi,
          fetchingUnits: false,
          fetchedUnits: true,
          fetchingAmiCharts: false,
          fetchedAmiCharts: true,
          fetchingAmiChartsError: undefined,
          fetchingUnitsError: undefined,
        }}
      >
        <ListingDetailsEligibility listing={testListing} imageSrc={"listing-eligibility.svg"} />
      </ListingDetailsContext.Provider>
    )

    expect(await findByText("Eligibility")).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
    done()
  })

  it("displays listing details eligibility section for a sales listing", async (done) => {
    axios.get.mockResolvedValue({ data: { preferences: defaultPreferences } })

    const { asFragment, findByText } = render(
      <ListingDetailsContext.Provider
        value={{
          units: unitsWithOneAmi,
          amiCharts: amiChartsWithOneAmi,
          fetchingUnits: false,
          fetchedUnits: true,
          fetchingAmiCharts: false,
          fetchedAmiCharts: true,
          fetchingAmiChartsError: undefined,
          fetchingUnitsError: undefined,
        }}
      >
        <ListingDetailsEligibility listing={openSaleListing} imageSrc={"listing-eligibility.svg"} />
      </ListingDetailsContext.Provider>
    )

    expect(await findByText("Eligibility")).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
    done()
  })

  it("displays listing details eligibility section for a listing with only SRO units", async (done) => {
    axios.get.mockResolvedValue({ data: { preferences: defaultPreferences } })
    const { asFragment, findByText } = render(
      <ListingDetailsContext.Provider
        value={{
          units: unitsWithOneAmi,
          amiCharts: amiChartsWithOneAmi,
          fetchingUnits: false,
          fetchedUnits: true,
          fetchingAmiCharts: false,
          fetchedAmiCharts: true,
          fetchingAmiChartsError: undefined,
          fetchingUnitsError: undefined,
        }}
      >
        <ListingDetailsEligibility
          listing={sroRentalListing}
          imageSrc={"listing-eligibility.svg"}
        />
      </ListingDetailsContext.Provider>
    )

    expect(await findByText("Eligibility")).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
    done()
  })

  it("displays listing details eligibility section for an SRO listing with expanded occupancy units", async (done) => {
    const listing = { ...sroRentalListing, Id: "a0W0P00000FIuv3UAD" }
    axios.get.mockResolvedValue({ data: { preferences: defaultPreferences } })

    const { asFragment, findByText } = render(
      <ListingDetailsContext.Provider
        value={{
          units: unitsWithOneAmi,
          amiCharts: amiChartsWithOneAmi,
          fetchingUnits: false,
          fetchedUnits: true,
          fetchingAmiCharts: false,
          fetchedAmiCharts: true,
          fetchingAmiChartsError: undefined,
          fetchingUnitsError: undefined,
        }}
      >
        <ListingDetailsEligibility listing={listing} imageSrc={"listing-eligibility.svg"} />
      </ListingDetailsContext.Provider>
    )

    expect(await findByText("Eligibility")).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
    done()
  })

  it("displays listing details eligibility section for an SRO listing with a mix of SRO units and non-SRO units", async (done) => {
    axios.get.mockResolvedValue({ data: { preferences: defaultPreferences } })

    const { asFragment, findByText } = render(
      <ListingDetailsContext.Provider
        value={{
          units: unitsWithOneAmi,
          amiCharts: amiChartsWithOneAmi,
          fetchingUnits: false,
          fetchedUnits: true,
          fetchingAmiCharts: false,
          fetchedAmiCharts: true,
          fetchingAmiChartsError: undefined,
          fetchingUnitsError: undefined,
        }}
      >
        <ListingDetailsEligibility
          listing={sroMixedRentalListing}
          imageSrc={"listing-eligibility.svg"}
        />
      </ListingDetailsContext.Provider>
    )
    expect(await findByText("Eligibility")).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
    done()
  })

  it("displays listing details eligibility section when habitat listing", async (done) => {
    axios.get.mockResolvedValue({ data: { preferences: defaultPreferences } })

    const { asFragment, findByText } = render(
      <ListingDetailsContext.Provider
        value={{
          units: unitsWithOneAmi,
          amiCharts: amiChartsWithOneAmi,
          fetchingUnits: false,
          fetchedUnits: true,
          fetchingAmiCharts: false,
          fetchedAmiCharts: true,
          fetchingAmiChartsError: undefined,
          fetchingUnitsError: undefined,
        }}
      >
        <ListingDetailsEligibility listing={habitatListing} imageSrc={"listing-eligibility.svg"} />
      </ListingDetailsContext.Provider>
    )

    expect(await findByText("Eligibility")).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
    done()
  })

  it("displays custom listing details check if you're eligible section for Shirley Chisholm listing 1", async (done) => {
    axios.get.mockResolvedValue({ data: { preferences: defaultPreferences } })

    const { asFragment, findByText } = render(
      <ListingDetailsContext.Provider
        value={{
          units: unitsWithOneAmi,
          amiCharts: amiChartsWithOneAmi,
          fetchingUnits: false,
          fetchedUnits: true,
          fetchingAmiCharts: false,
          fetchedAmiCharts: true,
          fetchingAmiChartsError: undefined,
          fetchingUnitsError: undefined,
        }}
      >
        <ListingDetailsEligibility
          listing={rentalEducatorListing}
          imageSrc={"listing-eligibility.svg"}
        />
      </ListingDetailsContext.Provider>
    )

    expect(
      await findByText(t("listings.customListingType.educator.eligibility.title"))
    ).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
    done()
  })

  it("displays ListingDetailsChisholmPreferences for educator listing 1", async (done) => {
    axios.get.mockResolvedValue({ data: { listings: [rentalEducatorListing] } })
    const { asFragment, findByText } = await renderAndLoadAsync(
      <ListingDetailsContext.Provider
        value={{
          units: unitsWithOneAmi,
          amiCharts: amiChartsWithOneAmi,
          fetchingUnits: false,
          fetchedUnits: true,
          fetchingAmiCharts: false,
          fetchedAmiCharts: true,
          fetchingAmiChartsError: undefined,
          fetchingUnitsError: undefined,
        }}
      >
        <ListingDetailsEligibility
          listing={rentalEducatorListing}
          imageSrc={"listing-eligibility.svg"}
        />
      </ListingDetailsContext.Provider>
    )

    expect(findByText(t("listings.customListingType.educator.preferences.part1"))).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
    done()
  })

  it("does not display ListingDetailsChisholmPreferences for a non-Chisholm listing", async (done) => {
    axios.get.mockResolvedValue({ data: { listings: [sroRentalListing] } })
    const { asFragment, findByText } = await renderAndLoadAsync(
      <ListingDetailsContext.Provider
        value={{
          units: unitsWithOneAmi,
          amiCharts: amiChartsWithOneAmi,
          fetchingUnits: false,
          fetchedUnits: true,
          fetchingAmiCharts: false,
          fetchedAmiCharts: true,
          fetchingAmiChartsError: undefined,
          fetchingUnitsError: undefined,
        }}
      >
        <ListingDetailsEligibility
          listing={sroRentalListing}
          imageSrc={"listing-eligibility.svg"}
        />
      </ListingDetailsContext.Provider>
    )

    expect(findByText(t("listings.customListingType.educator.preferences.part1"))).not.toBeNull()
    expect(asFragment()).toMatchSnapshot()
    done()
  })
})
