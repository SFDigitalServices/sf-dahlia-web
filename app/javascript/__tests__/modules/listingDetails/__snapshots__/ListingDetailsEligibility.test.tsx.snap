// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingDetailsLotteryInfo displays listing details eligibility section 1`] = `
<li
  className="responsive-content-item bg-primary-lighter"
>
  <header
    className="detail-header"
  >
    <img
      alt=""
      className="detail-header__image "
      src="listing-eligibility.svg"
    />
    <hgroup
      className="detail-header__hgroup"
    >
      <h2
        className="detail-header__title"
      >
        Eligibility
      </h2>
      <span
        className="detail-header__subtitle"
      >
        Income, occupancy, preferences, and subsidies
      </span>
      <span
        className="ui-icon ui-medium"
        data-test-id={null}
      >
        <svg
          fill="currentColor"
          viewBox="0 0 32 32"
        >
          <path
            d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
          />
        </svg>
      </span>
    </hgroup>
  </header>
  <li
    className="list-section custom-counter__item"
  >
    <header
      className="list-section__header custom-counter__header"
    >
      <hgroup>
        <h4
          className="custom-counter__title"
        >
          Household Maximum Income
        </h4>
        <span
          className="custom-counter__subtitle"
        >
          <div>
            <p
              className="mb-4"
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "For income calculations, household size includes everyone (all ages) living in the unit.",
                  }
                }
              />
            </p>
            <p
              className="mb-4"
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "People in your household may need <a href='https://sfmohcd.org/special-calculations-household-income' target='_blank'>special income calculations</a> if they:",
                  }
                }
              />
            </p>
            <ul
              className="list-disc ml-5"
            >
              <li>
                Are full-time students (but not the primary applicant).
              </li>
              <li>
                Receive non-taxable income (non-taxable income might include SSI, SSDI, child support payments, and workerâ€™s compensation benefits).
              </li>
            </ul>
          </div>
        </span>
      </hgroup>
    </header>
    <div
      style={
        Object {
          "overflowX": "auto",
        }
      }
    >
      <table
        className="w-full text-sm"
      >
        <thead>
          <tr>
            <th
              className=""
            >
              household size
            </th>
            <th
              className=""
            >
              maximum income per month
            </th>
            <th
              className=""
            >
              maximum income per year
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </li>
  <li
    className="list-section custom-counter__item"
  >
    <header
      className="list-section__header custom-counter__header"
    >
      <hgroup>
        <h4
          className="custom-counter__title"
        >
          Occupancy
        </h4>
        <span
          className="custom-counter__subtitle"
        >
          Occupancy limits for this building differ from household size, and do not include children under 6.
        </span>
      </hgroup>
    </header>
    <div
      style={
        Object {
          "overflowX": "auto",
        }
      }
    >
      <table
        className="w-full text-sm"
      >
        <thead>
          <tr>
            <th
              className=""
            >
              Unit Type
            </th>
            <th
              className=""
            >
              Occupancy
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </li>
  <li
    className="list-section custom-counter__item"
  >
    <header
      className="list-section__header custom-counter__header"
    >
      <hgroup>
        <h4
          className="custom-counter__title"
        >
          Lottery Preferences
        </h4>
        <span
          className="custom-counter__subtitle"
        >
          Anyone may enter the housing lottery for this listing. If your household has one of the following preferences, you will be considered in the order shown here. Each preference holder will be reviewed in lottery rank order.
        </span>
      </hgroup>
    </header>
    <div
      className="flex justify-center"
    >
      <span
        className="ui-icon ui-large spinner-animation"
        data-test-id={null}
      >
        <svg
          fill="currentColor"
          viewBox="0 0 100 100"
        >
          <circle
            cx="50"
            cy="50"
            fill="none"
            r="32"
            stroke="currentColor"
            strokeDasharray="150.79644737231007 52.26548245743669"
            strokeWidth="9"
          />
        </svg>
      </span>
    </div>
  </li>
  <li
    className="list-section custom-counter__item"
  >
    <header
      className="list-section__header custom-counter__header"
    >
      <hgroup>
        <h4
          className="custom-counter__title"
        >
          Rental Assistance
        </h4>
        <span
          className="custom-counter__subtitle"
        >
          Section 8 housing vouchers and other valid rental assistance programs can be used for this property. In the case of a valid rental subsidy, the required minimum income will be based on the portion of the rent that the tenant pays after use of the subsidy.
        </span>
      </hgroup>
    </header>
  </li>
  <li
    className="list-section custom-counter__item"
  >
    <header
      className="list-section__header custom-counter__header"
    >
      <hgroup>
        <h4
          className="custom-counter__title"
        >
          Additional Eligibility Rules
        </h4>
        <span
          className="custom-counter__subtitle"
        >
          Applicants must also qualify under the rules of the building.
        </span>
      </hgroup>
    </header>
    <div
      className="info-card"
    >
      <div
        className="info-card__header"
      >
        <h3
          className="info-card__title"
        >
          Credit History
        </h3>
      </div>
      <div
        className="expandable-text text-sm text-gray-700"
      >
         
        <div>
          <p>
            Provide a credit report with score from Equifax, Experian, or TransUnion dated within thirty (30) days of the application. 
          </p>
          <p>
            Accounts that are not current or that are derogatory will negatively affect the overall scoring, which could result in the denial of the application or an additional deposit may be required. 
          </p>
          <p>
            Collection accounts...
          </p>
        </div>
        <span
          className="button-toggle"
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="info-card"
    >
      <div
        className="info-card__header"
      >
        <h3
          className="info-card__title"
        >
          Rental History
        </h3>
      </div>
      <div
        className="expandable-text text-sm text-gray-700"
      >
         
        <div>
          <p>
            Provide minimum of 4 years rental history with at least two prior rentals in which you were responsible for paying the rent.  Applicants without rental history will still be considered. No Guarantors permitted. 
          </p>
          <p>
            Previous rental history will be reviewed and must exhibit no derogatory references. 
          </p>
          <p>
            Landlord references will only check for...
          </p>
        </div>
        <span
          className="button-toggle"
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="info-card"
    >
      <div
        className="info-card__header"
      >
        <h3
          className="info-card__title"
        >
          Criminal Background
        </h3>
      </div>
      <div
        className="expandable-text text-sm text-gray-700"
      >
         
        <span>
          Qualified applicants with criminal history will be considered for housing in compliance with Article 49 of the San Francisco Police Code: Fair Chance Ordinance.
        </span>
      </div>
    </div>
    <p>
      <a
        href="https://us.awp.autotask.net/1/filelink/113bd-37ed41a1-53a7f01459-2"
        target="_blank"
      >
        Find out more about Building Selection Criteria
      </a>
    </p>
  </li>
</li>
`;
